atuin/src/command/client/history.rs

```diff
@@ -349,37 +351,16 @@ impl Cmd {
         print0: bool,
         reverse: bool,
     ) -> Result<()> {
-        let session = if session {
-            Some(env::var("ATUIN_SESSION")?)
-        } else {
-            None
-        };
-        let cwd = if cwd {
-            Some(utils::get_current_dir())
-        } else {
-            None
+        let filters = match (session, cwd) {
+            (true, true) => [Session, Directory],
+            (true, false) => [Session, Global],
+            (false, true) => [Global, Directory],
+            (false, false) => [settings.filter_mode, Global],
```

Comment:
I don't really like this. Is there a clever way to allow passing either `&[one_thing]` or `&[two, things]` to a function that accepts `&[T]` without creating a vec?

Reply:
Not really - the slice is a reference to something, so the "something" needs to exist. A vec would be fine imo